code/datums/status_effects/debuffs.dm

```diff
@@ -67,8 +67,17 @@
 	return ..()
 
 /datum/status_effect/incapacitating/sleeping/tick()
-	if(owner.getStaminaLoss())
-		owner.adjustStaminaLoss(-0.5) //reduce stamina loss by 0.5 per tick, 10 per 2 seconds
+	if(owner.health > owner.crit_threshold)
+		var/healing = -0.2
+		if((locate(/obj/structure/bed) in owner.loc))
+			healing -= 0.3
+		else
+			if((locate(/obj/structure/table) in owner.loc))
+				healing -= 0.1
+		owner.adjustBruteLoss(healing)
+		owner.adjustFireLoss(healing)
+		owner.adjustToxLoss(healing)
```

Comment:
forced for slimies or this will do damage

Reply:
i think this does it